app/src/main/java/com/concordium/wallet/ui/cis2/send/SendTokenReceiptViewModel.kt

```diff
@@ -374,6 +380,24 @@ class SendTokenReceiptViewModel(
         transferRepository.insert(transfer)
     }
 
+    private fun saveRecipientToRecent(recipientAddress: String) = viewModelScope.launch {
+        val recentRecipient = recentRecipientRepository.getByAddress(recipientAddress)
+        recentRecipient?.let {
+            recentRecipientRepository.update(
+                it.copy(addedAt = System.currentTimeMillis())
+            )
+        } ?: run {
+            recentRecipientRepository.insert(
+                RecentRecipientEntity(
+                    id = 0,
+                    name = Account.getDefaultName(recipientAddress),
```

Comment:
The name must match the name of the address book entry if it is there. This also causes this recipient to stuck with whatever name it had when it was first added.
Join the data from the address book table when querying recent recepients instead of duplicating storing a name duplicate